src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -1084,189 +1084,192 @@ private void SaveServerCredentialsToCache(Uri uri, uint authScheme, ICredentials
                 return;
             }
 
-            try
+            using(state.CancellationToken.Register(() => state.Tcs.TrySetCanceled()))
```

Comment:
:grey_exclamation: delegate and closure allocation

    state.CancellationToken.Register(s => ((RequestState)s).Tcs.TrySetCanceled(), state)